crates/cairo-addons/src/vm/hint_definitions/precompiles.rs

```diff
@@ -118,3 +119,44 @@ pub fn modexp_output() -> Hint {
         },
     )
 }
+
+pub fn alt_bn128_pairing_check_hint() -> Hint {
+    Hint::new(
+        String::from("alt_bn128_pairing_check_hint"),
+        |vm: &mut VirtualMachine,
+         _exec_scopes: &mut ExecutionScopes,
+         ids_data: &HashMap<String, HintReference>,
+         ap_tracking: &ApTracking,
+         _constants: &HashMap<String, Felt252>|
+         -> Result<(), HintError> {
+            let data: Vec<u8> = serialize_sequence("data", vm, ids_data, ap_tracking)?
+                .iter()
+                .filter_map(|x| x.to_u8())
+                .collect();
+
+            // Give virtually infinite gas as we checked this in cairo before.
+            let result = run_pair(
+                &data,
+                pair::ISTANBUL_PAIR_PER_POINT,
+                pair::ISTANBUL_PAIR_BASE,
+                2u64.pow(64) - 1,
+            );
+
+            match result {
+                Ok(output) => {
+                    insert_value_from_var_name("error", 0, vm, ids_data, ap_tracking)?;
+
+                    let output = deserialize_sequence(output.bytes.to_vec(), vm)?;
+                    insert_value_from_var_name("output", output, vm, ids_data, ap_tracking)
+                }
+                Err(e) => {
+                    let error_string = e.to_string();
+                    let error_ascii = error_string.as_bytes();
+                    let error_ptr = vm.add_memory_segment();
+                    write_collection_to_addr(error_ptr, error_ascii, vm)?;
+                    insert_value_from_var_name("error", error_ptr, vm, ids_data, ap_tracking)
```

Comment:
could use directly map_or_else on run_pair instead of a match on assigned result